Common/src/main/resources/data/crafttweaker/scripts/crafttweaker_sequences_cauldron_saveddata.zs

```diff
@@ -0,0 +1,248 @@
+                        // if this sleep condition has ran before, lets return without running any other code in this function.
+                        return;
+                    }
+                    // check if it is raining
+                    var raining = level.raining;
+                    // If it is not raining, lets tell the player to wait for rain.
+                    if !raining {
+                        player.sendMessage("Wait for rain one more time!");
+                    } else {
+                        // If it is raining and this condition has passed, set that in the saved data
+                        savedData.put("third-" + pos.asLong(), true);
+                    }
+                })
+                .sleepUntil(level => {
+
+                    // Check if the saved data contains a key with the name "third-<cauldron position>" and if the value for that key is true.
+                    // The cauldron position is used to ensure that multiple cauldrons in the world can run this sequence at the same time.
+                    if(savedData.contains("third-" + pos.asLong()) && (savedData.getData<BoolData>("third-" + pos.asLong()) as BoolData) as bool) {
+                        // if this sleep condition has ran before, lets return true without having to wait for it again.
+                        return true;
+                    }
+                    // return if it was raining or not.
+                    return level.raining;
+                })
+                .then(level => {
+                    // Send a message to the player
+                    player.sendMessage("It's a terrible day for rain");
+
+                    // Reset the saved data so that this cauldron can be used for this sequence again.
+                    savedData.put("first-" + pos.asLong(), false);
+                    savedData.put("second-" + pos.asLong(), false);
+                    savedData.put("third-" + pos.asLong(), false);
+                })
+                // Start the sequence
+                .start();
+    }
+    // return InteractionResult.SUCCESS on the server, but CONSUME on the client.
+    return InteractionResult.sidedSuccess(cLevel.isClientSide);
+});
+
+
+// Sequence tasks can also use a SequenceContext, which can provide custom data to the sequence as well as allowing a sequence to be stopped.
+// The custom data provided to a sequence can be any type, but for the sake of this example, lets create a custom type.
+
+// Create a public class called 'MyObject'
+public class MyObject {
+
+    // A public string version that has a getter (we can call myObjectInstance.version)
+    public var version as string : get;
+
+    // A constructor so we can make our object.
+    public this(version as string) {
+        // Set the variable version above to the version passed into the constructor.
+        this.version = version;
+    }
+}
```

Comment:
Personally not a fan of names that are these generic in example scripts. I totally see some other script creating its own `MyObject`. Might I suggest `MySequenceContext` or `MySequenceData` instead?